lib/engines/content_block_manager/app/components/content_block_manager/shared/schedule_publishing_component.html.erb

```diff
@@ -96,13 +96,23 @@
     <% end %>
     </div>
     <div>
-      <%= render partial: "content_block_manager/content_block/shared/cancel_delete_button",
-                 locals: {
-                   url: content_block_manager.content_block_manager_content_block_edition_path(
-                     @content_block_edition,
-                     redirect_path: content_block_manager.content_block_manager_content_block_document_path(@content_block_edition.document),
-                     ),
-                 } %>
+      <% if is_rescheduling %>
+        <%= render "govuk_publishing_components/components/button", {
+          text: "Cancel",
+          name: "Cancel",
+          value: "Cancel",
+          href: back_link,
+          secondary_solid: true,
+        } %>
+      <% else %>
+        <%= render partial: "content_block_manager/content_block/shared/cancel_delete_button",
```

Comment:
Sorry to be a stickler, but I think this should be a component too, rather than rendering a partial. My instinct is that components should be able to stand on their own and not require knowledge of the actual application.

Reply:
you mean the `cancel_delete_button` should be a component, or the whole code that's wrapped in the `if` statement?

Reply:
I think the `cancel_delete_button` - referring to a partial outside of components feels like a bit of a smell

Reply:
It doesn't need to be massive, all it needs to accept is a URL argument I think. Then you can use it in the other places where the partial is used

Reply:
cool, done in two new commits 👍